sample-operators/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java

```diff
@@ -2,128 +2,102 @@
 
 import java.sql.Connection;
 import java.sql.DriverManager;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
 import java.sql.SQLException;
-import java.sql.Statement;
 import java.util.Base64;
+import java.util.Optional;
+
+import javax.cache.Cache;
+import javax.cache.CacheManager;
+import javax.cache.configuration.MutableConfiguration;
+import javax.cache.spi.CachingProvider;
 
 import org.apache.commons.lang3.RandomStringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import io.fabric8.kubernetes.api.model.OwnerReference;
 import io.fabric8.kubernetes.api.model.Secret;
 import io.fabric8.kubernetes.api.model.SecretBuilder;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.javaoperatorsdk.operator.api.reconciler.*;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.EventSourceRegistry;
+import io.javaoperatorsdk.operator.processing.event.source.polling.PerResourcePollingEventSource;
+import io.javaoperatorsdk.operator.sample.schema.Schema;
+import io.javaoperatorsdk.operator.sample.schema.SchemaService;
+
+import com.github.benmanes.caffeine.jcache.spi.CaffeineCachingProvider;
 
 import static java.lang.String.format;
 
 @ControllerConfiguration
-public class MySQLSchemaReconciler implements Reconciler<MySQLSchema> {
-  static final String USERNAME_FORMAT = "%s-user";
-  static final String SECRET_FORMAT = "%s-secret";
-
+public class MySQLSchemaReconciler
+    implements Reconciler<MySQLSchema>, ErrorStatusHandler<MySQLSchema>,
+    EventSourceInitializer<MySQLSchema> {
+  public static final String SECRET_FORMAT = "%s-secret";
+  public static final String USERNAME_FORMAT = "%s-user";
+  public static final int POLL_PERIOD = 500;
   private final Logger log = LoggerFactory.getLogger(getClass());
 
   private final KubernetesClient kubernetesClient;
   private final MySQLDbConfig mysqlDbConfig;
+  PerResourcePollingEventSource<Schema, MySQLSchema> perResourcePollingEventSource;
 
   public MySQLSchemaReconciler(KubernetesClient kubernetesClient, MySQLDbConfig mysqlDbConfig) {
     this.kubernetesClient = kubernetesClient;
     this.mysqlDbConfig = mysqlDbConfig;
   }
 
+  @Override
+  public void prepareEventSources(EventSourceRegistry<MySQLSchema> eventSourceRegistry) {
```

Comment:
But for now it was not receiving events if the schema was deleted or changed.

Reply:
OK so that maybe what I'm missing 😄 
Like I said, I'm not really familiar with the example so just looking at the code seemed like things were now more complex.

Reply:
I also need to think about how to integrate this new feature with the dependent resources work.

Reply:
or not sure what do you mean...